include/gatekeeper_flow.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Gatekeeper - DoS protection system.
+ * Copyright (C) 2016 Digirati LTDA.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _GATEKEEPER_FLOW_H_
+#define _GATEKEEPER_FLOW_H_
+
+#include <stdio.h>
+#include <stdint.h>
+
+struct ip_flow {
+	/* IPv4 or IPv6. */
+	uint16_t proto;
+
+	union {
+		struct {
+			uint32_t src;
+			uint32_t dst;
+		} v4;
+
+		struct {
+			uint8_t src[16];
+			uint8_t dst[16];
+		} v6;
+	} f;
+};
+
+uint32_t rss_ip_flow_hf(const void *data,
+	__attribute__((unused)) uint32_t data_len,
+	__attribute__((unused)) uint32_t init_val);
```

Comment:
I believe one can drop `__attribute__((unused))` at the prototype of a function. This would be best because if one ever needs the currently unused parameter, one only has to change at the definition of the function. If this is correct, please drop in the two cases above, and the one case below.